spring-data-jpa/src/main/java/org/springframework/data/jpa/repository/query/QueryEnhancerFactory.java

```diff
@@ -82,4 +73,56 @@ public static QueryEnhancer forQuery(DeclaredQuery query) {
 		}
 	}
 
+	/**
+	 * Get the native query enhancer for the given {@link DeclaredQuery query} based on {@link #NATIVE_QUERY_ENHANCER}.
+	 *
+	 * @param query the declared query.
+	 * @return new instance of {@link QueryEnhancer}.
+	 */
+	private static QueryEnhancer getNativeQueryEnhancer(DeclaredQuery query) {
+
+		if (NATIVE_QUERY_ENHANCER.equals(NativeQueryEnhancer.JSQL)) {
+			return new JSqlParserQueryEnhancer(query);
+		}
+		return new DefaultQueryEnhancer(query);
+	}
+
+	/**
+	 * Possible choices for the {@link #NATIVE_PARSER_PROPERTY}. Read current selection via {@link #select(ClassLoader)}.
+	 */
+	enum NativeQueryEnhancer {
+
+		AUTO, DEFAULT, JSQL;
+
+		static final String NATIVE_PARSER_PROPERTY = "spring.data.jpa.query.native.parser";
+
+		private static NativeQueryEnhancer from(@Nullable String name) {
+			if (!StringUtils.hasText(name)) {
+				return AUTO;
+			}
+			return NativeQueryEnhancer.valueOf(name.toUpperCase());
+		}
+
+		/**
+		 * @param classLoader ClassLoader to look up available libraries.
+		 * @return the current selection considering classpath avialability and user selection via
+		 *         {@link #NATIVE_PARSER_PROPERTY}.
+		 */
+		static NativeQueryEnhancer select(ClassLoader classLoader) {
+
+			if (!ClassUtils.isPresent("net.sf.jsqlparser.parser.JSqlParser", classLoader)) {
```

Comment:
For Graal Native, wouldn't we want to keep this check within a field?

Reply:
the check would still work just not be initialized at build time which is not necessary since its done only once and forgotten afterwards.